pkg/contracts/src/SafeArbitrator.sol

```diff
@@ -0,0 +1,136 @@
+
+    DisputeStruct[] public disputes; // Stores the dispute info. disputes[disputeID].
+    mapping(address arbitrable => address safe) public arbitrableTribunalSafe; //Map arbitrable address to tribunal safe address
+
+    error OnlySafe();
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Can only be called by the owner.");
+        _;
+    }
+
+    modifier onlySafe(address _arbitrable) {
+        if (msg.sender == arbitrableTribunalSafe[_arbitrable]) {
+            _;
+        } else {
+            revert OnlySafe();
+        }
+    }
+
+    /// @dev Constructor.
+    /// @param _arbitrationFee Amount to be paid for arbitration.
+    constructor(uint256 _arbitrationFee) {
+        arbitrationFee = _arbitrationFee;
+    }
+
+    /// @dev Set the arbitration fee. Only callable by the owner.
+    /// @param _arbitrationFee Amount to be paid for arbitration.
+    function setArbitrationFee(uint256 _arbitrationFee) external onlyOwner {
+        arbitrationFee = _arbitrationFee;
+    }
+
+    function registerSafe(address _safe) external {
+        arbitrableTribunalSafe[msg.sender] = _safe;
+    }
+
+    /// @inheritdoc IArbitrator
+    function createDispute(uint256 _choices, bytes calldata _extraData)
+        external
+        payable
+        override
+        returns (uint256 disputeID)
+    {
+        require(msg.value >= arbitrationCost(_extraData), "Arbitration fees: not enough.");
+        disputeID = disputes.length;
+        disputes.push(
+            DisputeStruct({
+                arbitrated: IArbitrable(msg.sender),
+                arbitratorExtraData: _extraData,
+                choices: _choices,
+                arbitrationFee: msg.value,
+                ruling: 0,
+                status: DisputeStatus.Waiting
+            })
+        );
+
+        emit DisputeCreation(disputeID, IArbitrable(msg.sender));
+    }
+
+    /// @inheritdoc IArbitrator
+    function createDispute(
+        uint256, /*_choices*/
+        bytes calldata, /*_extraData*/
+        IERC20, /*_feeToken*/
+        uint256 /*_feeAmount*/
+    ) external pure override returns (uint256) {
+        revert("Not supported");
+    }
+
+    /// @dev Give a ruling to a dispute.
+    /// @param _disputeID ID of the dispute to rule.
+    /// @param _ruling Ruling given by the arbitrator. Note that 0 means that arbitrator chose "Refused to rule".
+    /// @param _arbitrable Address of the arbitrable that the safe rules for".
+    function executeRuling(uint256 _disputeID, uint256 _ruling, address _arbitrable) external onlySafe(_arbitrable) {
+        DisputeStruct storage dispute = disputes[_disputeID];
+        require(_ruling <= dispute.choices, "Invalid ruling.");
+        require(dispute.status != DisputeStatus.Solved, "The dispute must not be solved.");
+
+        dispute.ruling = _ruling;
+        dispute.status = DisputeStatus.Solved;
+
+        (bool success,) = payable(msg.sender).call{value: dispute.arbitrationFee}("");
```

Comment:
I think that will be better to do it on the rule function of the CVstrategy, so that way the user that clicks on the rule when timeout has some economic incentive

Reply:
So yeah we discuss together and outcome was to let it that way if the Arbitrator is upgreadable to eventually add the withdraw ability.
Putting back to unresolved to keep track.

Reply:
yes have the upgredable stuff pending to do